File: Java/1817_Finding_the_Users_Active_Minutes.java
class Solution {
    public int[] findingUsersActiveMinutes(int[][] logs, int k) {
        int[] result = new int[k];
        Map<Integer, HashSet<Integer>> m = new HashMap<>();
        for(int i = 0; i < logs.length; i++){
            int user = logs[i][0];
            int min = logs[i][1];
            if(m.get(user) == null){
                HashSet<Integer> set = new HashSet<>();
                set.add(min);
                m.put(user, set);
            }
            else{
                Set<Integer> set = m.get(user);
                set.add(min);
            }
        }
        //System.out.println(m);
        for(Map.Entry<Integer, HashSet<Integer>> e : m.entrySet()){
            result[e.getValue().size()-1]++;
        }
        return result;
    }
}

File: CS/1817_Finding_the_Users_Active_Minutes.cs
public class Solution {
    public int[] FindingUsersActiveMinutes(int[][] logs, int k) {
        int[] result = new int[k];
        var m = new Dictionary<int, HashSet<int>>();
        
        foreach(var log in logs){
            if(!m.ContainsKey(log[0])){
                m.Add(log[0],new HashSet<int>(0));
            }
            m[log[0]].Add(log[1]);
        }
        
        foreach(var item in m){
            result[item.Value.Count - 1]++;
        }
        
        return result;
    }
}
